usi/src/main/java/daq/usi/UsiImpl.java

```diff
@@ -1,35 +1,35 @@
 package daq.usi;
 
+import static grpc.SwitchModel.OVS_SWITCH;
+
 import daq.usi.allied.AlliedTelesisX230;
 import daq.usi.cisco.Cisco9300;
 import daq.usi.ovs.OpenVSwitch;
-import grpc.Interface;
-import grpc.Power;
+import grpc.InterfaceResponse;
+import grpc.PowerResponse;
 import grpc.SwitchActionResponse;
 import grpc.SwitchInfo;
-import grpc.SwitchInput;
 import grpc.USIServiceGrpc;
 import io.grpc.stub.StreamObserver;
 import java.util.HashMap;
 import java.util.Map;
 
 public class UsiImpl extends USIServiceGrpc.USIServiceImplBase {
-  private Map<String, SwitchController> switchControllers;
+  private final Map<String, BaseSwitchController> switchControllers;
 
   public UsiImpl() {
     super();
     switchControllers = new HashMap<>();
   }
 
-  private ISwitchController getSwitchController(SwitchInput switchInput) {
-    if (!switchInput.hasSwitchInfo()) {
-      return new OpenVSwitch(switchInput.getFauxInterface());
+  private SwitchController getSwitchController(SwitchInfo switchInfo) {
+    if (switchInfo.getModel().equals(OVS_SWITCH)) {
```

Comment:
Still don't understand why OVS is special cased -- why isn't it just a generic drop-in replacement?  (If we discussed, can you please summarize the reason)

Reply:
OVS doesn’t need to be stored, doesn’t  implement the runnable interface, so it needs to be handled first.